PyInstaller/exceptions.py

```diff
@@ -11,4 +11,11 @@
 
 
 class ExecCommandFailed(SystemExit):
-    pass
+    def __init__(self, module_name, hook_module_name):
+        self.module_name = module_name
+        self.hook_module_name = hook_module_name
+    def hook_import_fail(self):
+        return f"""Failed to import module {self.module_name} required by hook for module {self.hook_module_name}.
+                Please check whether module {self.module_name} actually exists and whether the hook is 
+                compatible with your version of {self.hook_module_name}: You might want to read more about 
+                hooks in the manual and provide a pull-request to improve PyInstaller."""
```

Comment:
If you want to provide a template for an `Exception` then don't do it like this. Use a `classmethod`. Something like:

```python
class ImportErrorInHook(ImportError):
    # Leave __init__ alone.
    @classmethod
    def from_hook_names(cls, module_name, hook_module_name):
        return cls("""Failed to import {} in hook {}.  long template text yah deh dah"""
                   .format(module_name, hook_module_name))
```
Which you'd raise using:
```python
raise ImportErrorInHook.from_hook_names("foo", "hook-foo.py")
```

Reply:
The IMO correct way would be:
``` python
class HookError(Exception):
    "Base class for hook related errors"
    pass

class ImportErrorWhenRunningHook(HookError):
    def __str__(self):
        return ("Failed to import module %s …"
                 "… more text …"   # <- mind space at end of each part
                 "your version of %s …" % self.args)
```
then just
```
raise ImportErrorWhenRunningHook(self.module_name, self.hook_filename)
```

Of course, `ImportErrorInHook(HookError)` should be a separate exception.

Please test whether this behaves as expected then. If you want PyInstaller terminate without a traceback in case of such an exception, it micht be worth sub-classing both exceptions from SystemExit, too (`ImportErrorInHook(HookError, SystemExit)`), since this should suppress the traceback. Pleae give it a try.

Reply:
hi @htgoebel  - could you please explain your thinking regarding your comment "consecutive lines of multi-line string will printed with many spaces in front"? 

I believe you made this comment in the requested change below, but I cannot reply to it there. Thanks

Reply:
```
>>> print("""Failed to import module {self.module_name} required by hook for module {self.hook_module_name}.
...                 Please check whether module {self.module_name} actually exists and whether the hook is
...                 compatible with your version of {self.hook_module_name}: You might want to read more about
...                 hooks in the manual and provide a pull-request to improve PyInstaller.""")
Failed to import module {self.module_name} required by hook for module {self.hook_module_name}.
                Please check whether module {self.module_name} actually exists and whether the hook is
                compatible with your version of {self.hook_module_name}: You might want to read more about
                hooks in the manual and provide a pull-request to improve PyInstaller.
```

Reply:
Thanks - also, @htgoebel , why should `ImportErrorInHook(HookError)` be a separate exception when you advise raising `ImportErrorWhenRunningHook` for this issue?

Reply:
@RS357 We will likely want to generalise this in future to other exceptions happening inside of hooks. Currently we're only looking at `ImportError`s originating from hooks but we may for example also wish to replace `FileNotFoundError`s in hooks with a `FileNotFoundErrorInHook(HookError)`. Having this inheritance hierarchy allows us to use code like:
```python
try:
    [do something likely to raise an exception]
except HookError:
    # This would catch both ImportErrorInHook and FileNotFoundErrorInHook
    [respond appropriately]
```